opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/__init__.py

```diff
@@ -19,7 +19,7 @@
 from re import sub
 from shutil import which
 
-from pkg_resources import iter_entry_points
+from importlib.metadata import entry_points
```

Comment:
You are adding `importlib-metadata` to the dependencies, but this import is from the Python standard library. There is a discussion here: https://github.com/open-telemetry/opentelemetry-python/issues/3234 about using  `importlib-metadata` and its implications. Note my comment at the end.

Reply:
Hello, sorry for the delay. 
If I understand correctly, you mean having a standard library with its own constraint within Opentelemetry would be wrong. I also understand that the change of importlib-metadata version could be made later if an error is reported.

https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2181#discussion_r1504854087

Reply:
IMHO its fine to depend on `importlib-metadata`, as its the python-version independent backport of the stdlib feature.
it will be easier to develop against.